### load dataset

### load data from ../data/row/data_set.csv

In [ ]:
import sys

sys.path.append("../")

from src.data_loader import DataLoader
from src.data_explorer import DataExplorer

loader = DataLoader("../data/raw/data_set.csv")
df = loader.load()
loader.show_infos(df)





### Dataset information

- The dataset was succesfully loaded from 'data/raw/data_set.csv'

- The output shows that the dataset contains **1024** and **14 columns** 
- All columns have **1025 non-null values** , which means there no missing values detected      at   this stage. The dataset contain mainly integer variables , with one decimal varible oldpeak
-The next is to continues the initial exploration by displaying the  the first rows and last rows ,  checking duplicated rows and preparing the data cleaning process

# Exploring dataset

In [10]:
explorer = DataExplorer(df)
explorer.show_head(10)

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0
5,58,0,0,100,248,0,0,122,0,1.0,1,0,2,1
6,58,1,0,114,318,0,2,140,0,4.4,0,3,1,0
7,55,1,0,160,289,0,0,145,1,0.8,1,1,3,0
8,46,1,0,120,249,0,0,144,0,0.8,2,0,3,0
9,54,1,0,122,286,0,0,116,1,3.2,1,2,2,0


# First Rows Preview

The table at the top shows the first 10 rows of the dataset 
Each row represents one observation ,and each columns representes a medical varible related to heart sisease analysis
The dataset contains variables such as:
- 'age':age of patient
- 'sex':sex of patient
- 'cp':chest pain type
- 'trestbps':resting blood pressure
- 'chol':cholesterol level
- 'thalach':maximum heart rate achived
- 'oldpeak':ST depression value
- 'target':target variable used to indicate if patient has heart disease or not

In [11]:
explorer.show_tail(10)


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
1015,58,1,0,128,216,0,0,131,1,2.2,1,3,3,0
1016,65,1,3,138,282,1,0,174,0,1.4,1,1,2,0
1017,53,1,0,123,282,0,1,95,1,2.0,1,2,3,0
1018,41,1,0,110,172,0,0,158,0,0.0,2,0,3,0
1019,47,1,0,112,204,0,1,143,0,0.1,2,0,2,1
1020,59,1,1,140,221,0,1,164,1,0.0,2,0,2,1
1021,60,1,0,125,258,0,0,141,1,2.8,1,1,3,0
1022,47,1,0,110,275,0,0,118,1,1.0,1,1,2,0
1023,50,0,0,110,254,0,0,159,0,0.0,2,0,2,1
1024,54,1,0,120,188,0,1,113,0,1.4,1,1,3,0


### Last Rows Preview

The table above shows the last 10 rows of the dataset
This step helps verify that the dataset was loaded completely until the end of file which confirms the dataset format is consistent

In [12]:
explorer.show_columns()


['age',
 'sex',
 'cp',
 'trestbps',
 'chol',
 'fbs',
 'restecg',
 'thalach',
 'exang',
 'oldpeak',
 'slope',
 'ca',
 'thal',
 'target']

### Dataset Columns

This output lists all variables available in the dataset.

The dataset contains **14 columns**, including medical features and the `target` variable. These columns will be used in the next steps for data cleaning, analysis, and visualization.

In [13]:
explorer.missing_values()


age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64

# Missing Values
The missing values report shows '0' missing values for all columns. This means that no missing data was detected at this stage


In [15]:
explorer.duplicated_rows()


np.int64(723)

## Check duplicated rows

In this part we get **723** that's means there **723** duplicated rows in our DataSet

In [16]:
explorer.descriptive_statistics()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
count,1025.000000,1025.000000,1025.000000,1025.000000,1025.00000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000
mean,54.434146,0.695610,0.942439,131.611707,246.00000,0.149268,0.529756,149.114146,0.336585,1.071512,1.385366,0.754146,2.323902,0.513171
std,9.072290,0.460373,1.029641,17.516718,51.59251,0.356527,0.527878,23.005724,0.472772,1.175053,0.617755,1.030798,0.620660,0.500070
min,29.000000,0.000000,0.000000,94.000000,126.00000,0.000000,0.000000,71.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,48.000000,0.000000,0.000000,120.000000,211.00000,0.000000,0.000000,132.000000,0.000000,0.000000,1.000000,0.000000,2.000000,0.000000
50%,56.000000,1.000000,1.000000,130.000000,240.00000,0.000000,1.000000,152.000000,0.000000,0.800000,1.000000,0.000000,2.000000,1.000000
75%,61.000000,1.000000,2.000000,140.000000,275.00000,0.000000,1.000000,166.000000,1.000000,1.800000,2.000000,1.000000,3.000000,1.000000
max,77.000000,1.000000,3.000000,200.000000,564.00000,1.000000,2.000000,202.000000,1.000000,6.200000,2.000000,4.000000,3.000000,1.000000


# descriptive statistics

The table at the top presents the descriptive statistics of the dataset
It shows taht all columns contains **1024** , which confirm again that there is no missing values
The average age of patients is approximately **54**  with minimum age of **29** and max age of **77** 
The average resting blood pressure ('trestbps') is about ~**131.61**
This statistics give a first numerical overview of the dataset berfore moving to data cleaning and deeper analysis

In [17]:
explorer.overview()


DataSet shape: (1025, 14)

 Columns:
- age
- sex
- cp
- trestbps
- chol
- fbs
- restecg
- thalach
- exang
- oldpeak
- slope
- ca
- thal
- target
missing Values
age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64
Duplicated rows: 723


# General Overvieww Summary

This overview confirms the previous observations about the dataset structure

The most important point here is that the dataset contains **723** duplicated records

Therefore , the next notebook will focus mainly on removing duplicated rows and saving cleaned version of the data set
